GVRf/Framework/framework/src/main/java/org/gearvrf/GVRPicker.java

```diff
@@ -56,21 +58,51 @@
  * <li>onPick(GVRPicker)        called when the set of picked objects changes.</li>
  * <li>onNoPick(GVRPicker)      called once when nothing is picked.</li>
  * </ul
+ * Each cursor controller has an internal picker that generates pick events
+ * from that cursor. When a cursor is used, the pick events contain touch
+ * information. You can register as a listener for pick events from a specific cursor.
  * @see IPickEvents
  * @see GVRSceneObject#attachCollider(GVRCollider)
  * @see GVRCollider
  * @see GVRCollider#setEnable(boolean)
  * @see GVRPickedObject
+ * @see GVRCursorController#addPickEventListener(IEvents)
  */
-public class GVRPicker extends GVRBehavior {
+public class GVRPicker extends GVRBehavior implements IEventReceiver {
```

Comment:
Why should the base picker do so many new things now? Can't the base picker be used by or derived from a class that deals the controller stuff?

Reply:
The reason for IEventReceiver is that lots of objects need to observe pick and touch events. The original picker sent the events to the scene only but you can't have multiple listeners for the same event on the same target. Making the picker an event receiver allows it to have many objects listening to the same events. This change is independent of the controller although it makes working with the controller easier.
The picker now has the ability to generate touch events (which it could not before). I should have done this originally but at that time IO was owned by a different team. I think we should unify pick and touch events in 4.0 but I did not want to change the API in 3.3.
Each GVRCursorController now has its own picker and the logic to interact with that picker is in the controller.  One of our internal apps needs to be able to drive the picker independently from their own motion events which is why GVRPicker.processPick is exposed.

Reply:
The controller is in the picker for the sole reason events can be send. Users should not have to create a GVRPicker at all - it should be created for them automatically somewhere in the controller framework.

If the CursorController has its own picker why the Picker has a CursorController too?

Unify touch and pick events as in have them all in once place or provide higher level, more useful, events?

Still not convinced picking and android MotionEvents belong to the same unit. The very existence of processPick is a good sign it belongs to a more specialized entity.